src/Widgets/calendar/GridDay.vala

```diff
@@ -56,21 +57,30 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
 
         label = new Gtk.Label (null);
 
+        event_dots = new Gee.HashMap<string, Gtk.Widget> ();
```

Comment:
Why did you move these 4 lines here? Is it necessary?

Reply:
Needed to capture the events into the hashmap before doing the grid stuff for the days.